charts/skypilot/templates/api-deployment.yaml

```diff
@@ -176,6 +181,36 @@ spec:
         - name: gcp-config
           mountPath: /root/.config/gcloud
       {{- end }}
+      {{- if .Values.runpodCredentials.enabled }}
+      - name: create-runpod-credentials
+        {{- with .Values.securityContext }}
+        securityContext:
+          {{- toYaml . | nindent 10 }}
+        {{- end }}
+        image: {{ .Values.apiService.image }}
+        command: ["/bin/sh", "-c"]
+        args:
+        - |
+          echo "Setting up RunPod credentials..."
+          if [ -n "$RUNPOD_API_KEY" ]; then
+            echo "RunPod credentials found in environment variable."
+            mkdir -p /root/.runpod
+            echo "[default]" > /root/.runpod/config.toml
+            echo "api_key = \"$RUNPOD_API_KEY\"" >> /root/.runpod/config.toml
+          else
+            echo "RunPod credentials not found in environment variables. Skipping credentials setup."
+            sleep 600
```

Comment:
Why do we need to sleep here?

Reply:
I copied it from the AWS initContainer, with a mind to making as few changes to pre-existing chart as possible. I agree it doesn't make sense to sleep, since the pod will never even start if the secret doesn't correctly exist anyway. Happy to remove it.

Reply:
Ahh, got it. Did not notice there was also a sleep in aws, agreed that we keep it as is!

Reply:
cc @romilbhardwaj Do you know why we decide to sleep when there is no credentials? Is it for better surfacing the error msg to user?

Reply:
Yep! Without a sleep IIRC the echo would not get flushed to k8s logs. Though yeah I think the sleep can be safely reduced.